### ISSUE TITLE
Unknown linux distro Pop

### ISSUE THREAD

**Issue by estevaoem** [nodkz/mongodb-memory-server#468]
Whenever I try to run jest, I get this output from the terminal:

> Determining test suites to run...Unknown linux distro Pop
> Falling back to legacy MongoDB build!
> Starting the instance failed, please enable debug for more infomation
> 
> 
>   ● Test suite failed to run
> 
>     Status Code is 403 (MongoDB's 404)
>     This means that the requested version-platform combination doesn't exist
> 
>       at ClientRequest.<anonymous> (node_modules/mongodb-memory-server-core/lib/util/MongoBinaryDownload.js:378:44)

Also, the same happens with Kde Neon, but it shows me `Unknown linux distro Neon`

**Comment by hasezoey** [comment]
your distro is "not supported", but if its bases on something (like ubuntu / debian / arch), then most likely there is an `/etc/upstream/lsb-release`, which is check since the 7.0.0 beta version

TL;DR: please try an recent beta version of this package (currently: `7.0.0-beta.38`) [this is an breaking version, some documentation might not be updated / relevant anymore]

**Comment by estevaoem** [comment]
Sorry for the misslabeling.  And sorry if my question sems dumb, but do I have to compile it? I'm asking this because I've added     `"mongodb-memory-server": "^7.0.0-beta.38",` to my package.json and npm installed `mongodb-memory-server@6.6.7`

**Comment by hasezoey** [comment]
> Sorry for the misslabeling

i just did it, because i knew what the problem was about, and so knew it wasnt an problem (anymore)

> And sorry if my question sems dumb, but do I have to compile it

no its on npm

---

if you have problem i recommend deleting `node_modules` and try to install it again, and if `npm` still dosnt want, try `yarn` (at least to try if it works for that version)

**Comment by gregbacchus** [comment]
To fix this we need a small change to `packages/mongodb-memory-server-core/src/util/MongoBinaryDownloadUrl.ts`

```typescript
  getLinuxOSVersionString(os: LinuxOS): string {
    if (/(ubuntu|Pop)/i.test(os.dist)) {
      return this.getUbuntuVersionString(os);
    } else if (/elementary OS/i.test(os.dist)) {
```

**Comment by hasezoey** [comment]
@gregbacchus like i said in an earlier comment, this should already be fixed in the 7.0.0 beta, so please try that

**Comment by hasezoey** [comment]
closing because fixed in `7.0.0-beta.16`
